internal/run/convert.go

```diff
@@ -0,0 +1,104 @@
+package run
+
+import (
+	"errors"
+	"fmt"
+	"log"
+
+	convertcmd "github.com/helm/helm-2to3/cmd"
+	"github.com/helm/helm-2to3/pkg/common"
+	"github.com/pelotech/drone-helm3/internal/env"
+	"helm.sh/helm/v3/pkg/action"
+	"helm.sh/helm/v3/pkg/cli"
+	"helm.sh/helm/v3/pkg/storage/driver"
+)
+
+func v3ReleaseFound(release string, cfg *action.Configuration) bool {
+
+	// check if release exists
+	_, err := cfg.Releases.Deployed(release)
+	switch {
+	// case errors.Is(err, driver.ErrReleaseNotFound), errors.Is(err, driver.ErrNoDeployedReleases):
+	case errors.Is(err, driver.ErrReleaseNotFound):
+		log.Printf("No v3 Release of %s found", release)
+	case err == nil:
+		log.Printf("A v3 Release of %s was found", release)
+		return true
+	}
+
+	return false
+}
+
+// Convert holds the parameters to run the Convert action
+type Convert struct {
+	namespace      string
+	debug          action.DebugLog
+	kubeConfig     string
+	kubeContext    string
+	convertOptions convertcmd.ConvertOptions
+}
+
+// NewConvert initialize Convert by using values from env.Config
+func NewConvert(cfg env.Config, kubeConfig string, kubeContext string) *Convert {
+
+	convert := &Convert{
+		namespace:   cfg.Namespace,
+		kubeConfig:  kubeConfig,
+		kubeContext: kubeContext,
+	}
+
+	if cfg.ReleaseVersionsMax == 0 {
```

Comment:
**note:** This changes the behavior of `0` meaning "convert all releases", which is probably not a huge issue since you can [set an arbitrary value higher than the total releases to get the same behavior](https://github.com/helm/helm-2to3/blob/master/cmd/convert.go#L126), but worth noting.

Also, it sounds like this also means it will likely [leave some garbage around](https://github.com/helm/helm-2to3/blob/master/cmd/convert.go#L131) after the conversion that needs to be manually cleaned up.

Reply:
yeah, it is weird to check if int is 0 as there's no nil value, can probably create a custom object or mark it as a pointer, but don't know if that will have other consecuenses.

Do you have any suggestion in how to handle this?

Reply:
I was thinking at some point in simply hardcoding this and don't expose it as part of the plugin interface

Reply:
It's fine as is, just noting that this will almost certainly leave around a bunch of useless configmaps. I guess one option is to also have this run a `cleanup`?

Reply:
I think running cleanup makes sense if `delete_v2_releases` is set, that would be our default value.